core/src/main/scala/org/apache/spark/deploy/history/HistoryAppStatusStore.scala

```diff
@@ -63,8 +63,25 @@ private[spark] class HistoryAppStatusStore(
     }
   }
 
+  val STDOUT_URL_GROUPS = conf.get(LIVE_EXECUTOR_LOR_URL_CAPTURING_GROUPS)
+    .map(_.split(","))
+    .getOrElse(Array.empty[String])
+
+  val STDOUT_URL_REGEX = conf.get(LIVE_EXECUTOR_LOG_URL_REGEX)
+    .map(r => new Regex(r, STDOUT_URL_GROUPS: _*))
+
   private def replaceLogUrls(exec: v1.ExecutorSummary, urlPattern: String): v1.ExecutorSummary = {
-    val attributes = exec.attributes
+    val attributes = STDOUT_URL_REGEX.flatMap { regEx =>
+      exec.executorLogs
+        .get("stdout")
+        .flatMap(regEx.findFirstMatchIn)
+        .map { urlMatch =>
+          STDOUT_URL_GROUPS.foldLeft(exec.attributes) { (attribs, groupName) =>
```

Comment:
I don't think this block should be executed at all when `exec.attributes` is not empty; IMO this should only apply to legacy event logs that don't record that information.

Reply:
+1 Let's just use `attributes` if it exists. I can see the case this helps even if attributes are presented: the number of `attributes` are updated which new attributes can be extracted from URL, but at least with current YARN URL we have extracted almost everything so it will unlikely happen.